src/features/codeActionProvider.ts

```diff
@@ -13,26 +13,23 @@ import { Options } from '../omnisharp/options';
 import { FileModificationType } from '../omnisharp/protocol';
 import { Uri } from 'vscode';
 import CompositeDisposable from '../CompositeDisposable';
+import { Observable } from 'rxjs/Observable';
+import Disposable from '../Disposable';
 
 export default class CodeActionProvider extends AbstractProvider implements vscode.CodeActionProvider {
 
     private _options: Options;
     private _commandId: string;
 
-    constructor(server: OmniSharpServer) {
+    constructor(server: OmniSharpServer, optionStream: Observable<Options>) {
         super(server);
 
         this._commandId = 'omnisharp.runCodeAction';
 
-        this._resetCachedOptions();
-
-        let d1 = vscode.workspace.onDidChangeConfiguration(this._resetCachedOptions, this);
-        let d2 = vscode.commands.registerCommand(this._commandId, this._runCodeAction, this);
-        this.addDisposables(new CompositeDisposable(d1, d2));
-    }
-
-    private _resetCachedOptions(): void {
         this._options = Options.Read(vscode);
+        let configChangedDisposable = new Disposable(optionStream.subscribe(options => this._options = options)); 
```

Comment:
Do we actually need to make this a disposable now? Are we not allowed to dispose the underlying observable if there are still observers?